### PR TITLE
Singlecell color columns

### PR DESCRIPTION
**Pull request by airenzp**
## Description

Supported coloring by different columns in the single cell plot. See corresponding [PR](https://github.com/stjude/sjpp/pull/501) in sjpp and in [GDC](https://github.com/stjude/ppgdc/pull/2). Do we want to unitfy the plots that are the same with a different color column into one plot with mulitple color columns, like in the BALL-scrna? Please check.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Comment by airenzp** [comment]
This is the alternative I mean


https://github.com/user-attachments/assets/780886ac-dea6-4e8e-bc8d-6782cc1d88aa

File: client/plots/singleCellPlot.js [hunk 1]
```diff
@@ -121,69 +121,63 @@ class singleCellPlot {
...
 						})
 					})
-				headerDiv.append('label').text(plot.name).attr('for', `show${id}`)
+				topDiv.append('label').text(plot.name).attr('for', `show${id}`)
 			}
+		let selectCategory, violinBt, geneSearch, colorByGene, colorByPlot, searchboxDiv
 		if (q.singleCell?.geneExpression) {
-			searchGeneDiv.append('label').html('Gene expression:').style('padding', '10px')
-			const geneSearch = addGeneSearchbox({
+			headerDiv.append('label').text('Color by:').style('margin-right', '5px')
```
Comment (xzhou82) [inline client/plots/singleCellPlot.js:128]: please use `dom/radiobutton.js` to make radios which uses random id and is safe against conflict with external portals
on checking `Gene exp` radio button, apply focus on gene search box and no need for user having to click on it

**Review by robinpaul85 — APPROVED** [review]

File: client/plots/singleCellPlot.js [hunk 2]
```diff
@@ -414,7 +434,20 @@ class singleCellPlot {
...
 		copyMerge(this.settings, this.config.settings.singleCellPlot)
-
+		this.colorByDivs = []
+		// Only add unique colorColumn among plots as option
+		const uniqueColorColumns = new Set()
+		if (this.dom.selectCategory) {
+			this.dom.selectCategory.selectAll('*').remove()
+			for (const plot of this.state.termdbConfig?.queries.singleCell.data.plots) {
+				const colorColumn = this.state.config.colorBy?.[plot.name] || plot.colorColumns[0]
+				const display = this.settings[`show${plot.name.replace(/\s+/g, '')}`]
```
Comment (xzhou82) [inline client/plots/singleCellPlot.js:444]: i wonder what this is, may comment on its purpose at the beginning comment section

Comment (airenzp) [reply client/plots/singleCellPlot.js]: solved

File: client/plots/singleCellPlot.js (hunk 1 above)
Comment (airenzp) [reply client/plots/singleCellPlot.js:130]: Used make_radios. Tried focus() on input but not working

**Review by xzhou82 — APPROVED** [review]
thanks
somehow the gene search box auto focus doesn't work. in the termtypeselect it does work. should look into it further
also worth to add more comment on the singlecell plot state and plot properties

**Pull request merged**